### PR TITLE
Feat/server get restaurant route

### PR DESCRIPTION
**Pull request by tmkuba**
https://trello.com/c/ku4E5Cg8

### CODE REVIEW

**Review by leungnamson1023 — COMMENTED** [review]
nice Mark, looks good. Maybe it's just me though but I thought the first describe was the same as the second describe func but I later realized that it has a different endpoint. Could be just me though, but maybe just a slight more detail regarding the difference between the first describe func and the second describe func. Could be just me being not too familiar with the appropriate styling/syntax for testing

**Review by matgbx — APPROVED** [review]
looks great, nice work!

File: server/routers/restaurant.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      time: 20,
+      remaining: 5,
+    },
+    {
+      time: 21,
+      remaining: 14,
+    },
+    ],
+  };
+  res.status(200).send(response);
```
Comment (matgbx) [inline server/routers/restaurant.js]: beth - is status 200 necessary

File: server/app.js [hunk 2]
```diff
@@ -2,10 +2,14 @@ const path = require('path');
...
 
+const restaurantRouter = require('./routers/restaurant');
+
 const app = express();
 
 // middleware
 app.use(morgan('dev'));
+
 app.use('/', express.static(path.join(__dirname, '../client/dist')));
+app.use('/restaurant', restaurantRouter);
```
Comment (matgbx) [inline server/app.js]: beth - why singular restaurant and not restaurants?

File: server/routers/restaurant.test.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    test('should return 200 response', () => request(app).get('/restaurant/300/reservations/')
+      .then((response) => {
+        expect(response.statusCode).toBe(200);
+      }));
+
+    test('should return expected object shape', () => request(app).get('/restaurant/300/reservations/')
+      .then((response) => {
+        const responseKeys = Object.keys(response.body);
+
+        expect(responseKeys.length).toBe(2);
```
Comment (matgbx) [inline server/routers/restaurant.test.js]: beth - see if there is a nicer way to do this in jest if not consider other expect libraries

File: server/routers/restaurant.test.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        expect(reservationObjKeys.length).toBe(2);
+        expect(reservationObjKeys.includes('time')).toBe(true);
+        expect(reservationObjKeys.includes('remaining')).toBe(true);
+
+        expect(typeof response.body.reservations[0].time).toBe('number');
+        expect(typeof response.body.reservations[0].remaining).toBe('number');
+      }));
+  });
+
+  describe('GET /restaurant/:id/reservations/2018-03-01', () => {
```
Comment (matgbx) [inline server/routers/restaurant.test.js]: beth - consider a way to generate dates for this test

**Review by matgbx — CHANGES_REQUESTED** [review]
the only thing I think is blocking is /restaurant - it should be plural - everything else is your call

**Pull request merged**